src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/external_repacking_adjuster.cpp

```diff
@@ -14,59 +14,144 @@
-    auto& optimal_descs = cpu_config->m_in_requested_descs;
-    for (const auto& i : m_param_idces_with_external_repacking) {
+
+    size_t data_size = 0;
+    for (const auto& p : m_executors) {
+        const auto& i = p.first;
         const auto& shape = cpu_config->io_shapes[i];
-        const auto& K = *++shape.rbegin();
-        const auto& N = *shape.rbegin();
-
-        const auto& precision = linear_ir.get_parameters()[i]->get_node()->get_output_element_type(0);
-        const auto vnni_factor = brgemm_utils::compute_vnni_factor(precision);
-        const size_t brgemm_kernel_rank = 2;
-        // Firstly, batch dims are set
-        VectorDims requested_blocked_shape(shape.begin(), shape.end() - brgemm_kernel_rank);
-        // Then, the blocked dims are formed
-        requested_blocked_shape.insert(requested_blocked_shape.end(),
-                                       {snippets::utils::div_up(K, vnni_factor),
-                                        std::max(N, brgemm_utils::repacking::compute_inner_n_block(precision)),
-                                        vnni_factor});
-
-        VectorDims requested_order(shape.size() - brgemm_kernel_rank);
-        std::iota(requested_order.begin(), requested_order.end(), 0);
-        const auto last_idx = shape.size() - 1;
-        requested_order.insert(requested_order.end(), {last_idx - 1, last_idx, last_idx - 1});
-
-        optimal_descs[i] =
-            std::make_shared<CpuBlockedMemoryDesc>(precision, Shape(shape), requested_blocked_shape, requested_order);
-
-        ov::snippets::VectorDims shape_for_offset(cpu_config->tensor_rank - shape.size(), 1);
-        shape_for_offset.insert(shape_for_offset.end(), requested_blocked_shape.begin(), requested_blocked_shape.end());
-        m_configurator->compute_offsets(shape_for_offset, i, 0);
+
+        const auto& layout = cpu_config->io_layouts[i];
+        const auto planar_shape = ov::snippets::utils::get_planar_vdims(shape, layout);
+        const auto& K = *++planar_shape.rbegin();
+        const auto& N = *planar_shape.rbegin();
+
+        const auto& prc = linear_ir.get_parameters()[i]->get_node()->get_output_element_type(0);
+        const auto blk_shape = get_blk_shape(planar_shape, prc);
+
+        // src data + dst data per kernel call
+        const auto src_data = N * K * prc.size();
+        const auto dst_data =
+            std::accumulate(blk_shape.rbegin(), blk_shape.rbegin() + 3, prc.size(), std::multiplies<size_t>());
+        data_size += src_data + dst_data;
+
+        update_kernel(p.second, shape, layout, N, K, prc);
     }
+
+    const auto L2_cache_size = dnnl::utils::get_cache_size(2, true);
+    const auto fit_into_L2 = data_size < L2_cache_size;
+    // Heuristic: If external repacking data doesn't fit in the cache L2,
+    //            external repacking should be executed in seperate parallel section before kernel execution.
```

Comment:
Why do we ignore L1 cache here? It's also used to store the data and it's size might affect optimal RepackingImplType